File: app/src/main/java/com/simon/harmonichackernews/data/RepoInfo.java
package com.simon.harmonichackernews.data;

public class RepoInfo {

    public String name;
    public String owner;
    public String about;
    public String website;
    public String license;
    public String language;
    public int stars;
    public int watching;
    public int forks;

    public String formatStars() {
        return LinkPreviewFormatUtils.formatCount(stars, "star", "stars");
    }

    public String formatWatching() {
        return LinkPreviewFormatUtils.kFormat(watching) + " watching";
    }

    public String formatForks() {
        return LinkPreviewFormatUtils.formatCount(forks, "fork", "forks");
    }

    public String getShortenedUrl() {
        return LinkPreviewFormatUtils.shortenUrl(website);
    }

}

File: core/src/commonMain/kotlin/com/simon/harmonichackernews/data/RepoInfo.kt
package com.simon.harmonichackernews.data

import kotlinx.serialization.Serializable

@Serializable
data class RepoInfo(
    val name: String? = null,
    val owner: String? = null,
    val avatarUrl: String? = null,
    val about: String? = null,
    val website: String? = null,
    val license: String? = null,
    val language: String? = null,
    val stars: Int = 0,
    val watching: Int = 0,
    val forks: Int = 0,
) {
    fun formatStars(): String = LinkPreviewFormatUtils.formatCount(stars, "star", "stars")
    fun formatWatching(): String = "${LinkPreviewFormatUtils.kFormat(watching)} watching"
    fun formatForks(): String = LinkPreviewFormatUtils.formatCount(forks, "fork", "forks")
    val shortenedUrl: String? get() = LinkPreviewFormatUtils.shortenUrl(website)
}
